wordpress-mobile/WordPress-Aztec-iOS
# Issue/77 image upload support

Refs #77 

Adds support to image/attachment uploads on the editor

How to test:
 - Launch the demo app
 - Select the editor demo option
 - Add media elements to the editor
 - Check if the upload progress is being done to the end correctly, and image returns to valid state
 - Rotate, scroll and move around the editor while the upload/progress is ongoing
  - Exit the app go to Xcode and edit the variable `mediaErrorMode` on the `EditorDemoController` to `true`
  - Add a new media element, check if the progress goes up to 25% and then fails with a message
  - Repeats the other tests mentioned above.

Notes:
 - This still not supports switching to HTML mode and back to native while the upload is ongoing.
